muDOT zoo example takes a long (infinite?) time to check

`zoo.scala` is as follows:

``` scala
object Test {
type Meat = {
  type IsMeat = Any
}
type Grass = {
  type IsGrass = Any
}
type Animal = {
  type Food
  def eats(food: Food): Unit
  def gets: Food
}
type Cow = {
  type IsMeat = Any
  type Food <: Grass
  def eats(food: Grass): Unit
  def gets: Grass
}
type Lion = {
  type Food = Meat
  def eats(food: Meat): Unit
  def gets: Meat
}
def newMeat: Meat = new {
  type IsMeat = Any
}
def newGrass: Grass = new {
  type IsGrass = Any
}
def newCow: Cow = new {
  type IsMeat = Any
  type Food = Grass
  def eats(food: Grass) = ()
  def gets = newGrass
}
def newLion: Lion = new {
  type Food = Meat
  def eats(food: Meat) = ()
  def gets = newMeat
}
val milka = newCow
val leo = newLion
leo.eats(milka)
}
```

`scalac zoo.scala` fails with:

``` bash
zoo.scala:10: error: Parameter type in structural refinement may not
refer to a type member of that refinement
  def eats(food: Food): Unit
```

while `dottyc zoo.scala` just seems to hang.

--- a/src/dotty/tools/dotc/transform/Splitter.scala
+++ b/src/dotty/tools/dotc/transform/Splitter.scala
@@ -70,7 +70,7 @@ class Splitter extends TreeTransform {
     }
 
     def isStructuralSelect(tp: Type): Boolean = tp.stripTypeVar match {
-      case tp: RefinedType => tp.refinedName == name || isStructuralSelect(tp)
+      case tp: RefinedType => tp.refinedName == name || isStructuralSelect(tp.parent)
       case tp: TypeProxy => isStructuralSelect(tp.underlying)
       case AndType(tp1, tp2) => isStructuralSelect(tp1) || isStructuralSelect(tp2)
       case _ => false

--- a/test/dotc/tests.scala
+++ b/test/dotc/tests.scala
@@ -75,6 +75,7 @@ class tests extends CompilerTest {
       defaultOptions = noCheckOptions)
         // -Ycheck fails because there are structural types involving higher-kinded types.
         // these are illegal, but are tested only later.
+  @Test def neg_zoo = compileFile(negDir, "zoo", xerrors = 1)
   @Test def neg_t1192_legalPrefix = compileFile(negDir, "t1192", xerrors = 1)
   @Test def neg_tailcall_t1672b = compileFile(negDir, "tailcall/t1672b", xerrors = 6)
   @Test def neg_tailcall_t3275 = compileFile(negDir, "tailcall/t3275", xerrors = 1)

--- a/tests/neg/zoo.scala
+++ b/tests/neg/zoo.scala
@@ -1,0 +1,44 @@
+object Test {
+type Meat = {
+  type IsMeat = Any
+}
+type Grass = {
+  type IsGrass = Any
+}
+type Animal = {
+  type Food
+  def eats(food: Food): Unit
+  def gets: Food
+}
+type Cow = {
+  type IsMeat = Any
+  type Food <: Grass
+  def eats(food: Grass): Unit
+  def gets: Grass
+}
+type Lion = {
+  type Food = Meat
+  def eats(food: Meat): Unit
+  def gets: Meat
+}
+def newMeat: Meat = new {
+  type IsMeat = Any
+}
+def newGrass: Grass = new {
+  type IsGrass = Any
+}
+def newCow: Cow = new {
+  type IsMeat = Any
+  type Food = Grass
+  def eats(food: Grass) = ()
+  def gets = newGrass
+}
+def newLion: Lion = new {
+  type Food = Meat
+  def eats(food: Meat) = ()
+  def gets = newMeat
+}
+val milka = newCow
+val leo = newLion
+leo.eats(milka) // structural select not supported
+}

--- a/tests/pos/zoo.scala
+++ b/tests/pos/zoo.scala
@@ -1,0 +1,44 @@
+object Test {
+type Meat = {
+  type IsMeat = Any
+}
+type Grass = {
+  type IsGrass = Any
+}
+type Animal = {
+  type Food
+  def eats(food: Food): Unit
+  def gets: Food
+}
+type Cow = {
+  type IsMeat = Any
+  type Food <: Grass
+  def eats(food: Grass): Unit
+  def gets: Grass
+}
+type Lion = {
+  type Food = Meat
+  def eats(food: Meat): Unit
+  def gets: Meat
+}
+def newMeat: Meat = new {
+  type IsMeat = Any
+}
+def newGrass: Grass = new {
+  type IsGrass = Any
+}
+def newCow: Cow = new {
+  type IsMeat = Any
+  type Food = Grass
+  def eats(food: Grass) = ()
+  def gets = newGrass
+}
+def newLion: Lion = new {
+  type Food = Meat
+  def eats(food: Meat) = ()
+  def gets = newMeat
+}
+val milka = newCow
+val leo = newLion
+//leo.eats(milka) // structural select not supported
+}

